src/controllers/music/songs.js

```diff
@@ -8,197 +8,201 @@ import * as userSettings from '../../scripts/settings/userSettings';
-                        id: 'Runtime,AlbumArtist,Album,SortName'
-                    }],
-                    callback: function () {
-                        getQuery(tabContent).StartIndex = 0;
-                        reloadItems(tabContent);
-                    },
-                    query: getQuery(tabContent),
-                    button: e.target
-                });
+            Events.on(filterDialog, 'filterchange', function () {
+                getQuery(tabContent).StartIndex = 0;
+                reloadItems(tabContent);
             });
-        }
-
-        initPage(tabContent);
+            filterDialog.show();
+        });
+    };
+
+    function shuffle() {
+        ApiClient.getItem(ApiClient.getCurrentUserId(), params.topParentId).then(function (item) {
+            playbackManager.shuffle(item);
+        });
+    }
 
-        self.renderTab = function () {
-            reloadItems(tabContent);
-        };
+    self.getCurrentViewStyle = function () {
+        return getPageData(tabContent).view;
+    };
+
+    function initPage(tabContent) {
+        tabContent.querySelector('.btnFilter').addEventListener('click', function () {
+            self.showFilterMenu();
+        });
+        tabContent.querySelector('.btnSort').addEventListener('click', function (e) {
+            libraryBrowser.showSortMenu({
+                items: [{
+                    name: globalize.translate('OptionTrackName'),
+                    id: 'Name'
+                }, {
+                    name: globalize.translate('Album'),
+                    id: 'Album,SortName'
+                }, {
+                    name: globalize.translate('AlbumArtist'),
+                    id: 'AlbumArtist,Album,SortName'
+                }, {
+                    name: globalize.translate('Artist'),
+                    id: 'Artist,Album,SortName'
+                }, {
+                    name: globalize.translate('OptionDateAdded'),
+                    id: 'DateCreated,SortName'
+                }, {
+                    name: globalize.translate('OptionDatePlayed'),
+                    id: 'DatePlayed,SortName'
+                }, {
+                    name: globalize.translate('OptionPlayCount'),
+                    id: 'PlayCount,SortName'
+                }, {
+                    name: globalize.translate('OptionReleaseDate'),
+                    id: 'PremiereDate,AlbumArtist,Album,SortName'
+                }, {
+                    name: globalize.translate('Runtime'),
+                    id: 'Runtime,AlbumArtist,Album,SortName'
+                }],
+                callback: function () {
+                    getQuery(tabContent).StartIndex = 0;
+                    reloadItems(tabContent);
+                },
+                query: getQuery(tabContent),
+                button: e.target
+            });
+            tabContent.querySelector('.btnShuffle').addEventListener('click', shuffle);
```

Comment:
Move this out of block - that's why it doesn't work until you pressed the sort button